Added ccx library

dsjen opened the pull request:
* Refactored tests to use a variety of course IDs using ddt.

@brianhw, this is needed to test out CCX on stage.

thallada commented:
I restarted the build that failed because of elasticsearch flakiness.

dsjen commented:
Ugh, thanks!  Looks like it passed. :)

thallada commented on analytics_data_api/v0/tests/views/test_learners.py:
```diff
@@ -641,12 +641,16 @@ def test_csv_fields(self, fields, valid_fields):
+    @classmethod
+    def setUpClass(cls):
+        super(CourseLearnerMetadataTests, cls).setUpClass()
+        cls.course_id = 'edX/DemoX/Demo_Course'
```
Is there a reason you can't use `CourseSamples.course_ids` here like you were doing everywhere else?

dsjen replied:
Ugh, I was getting lazy!  Updated.  :)

thallada commented on requirements/base.txt:
```diff
@@ -13,4 +13,6 @@ ordered-set==2.0.1                  # MIT
 
 -e git+https://github.com/edx/opaque-keys.git@d45d0bd8d64c69531be69178b9505b5d38806ce0#egg=opaque-keys
+# custom opaque-key implementations for ccx
+git+https://github.com/jazkarta/ccx-keys.git@e6b03704b1bb97c1d2f31301ecb4e3a687c536ea#egg=ccx-keys
```
Hmm... this repo is titled "obsolete-ccx-keys" now. Any reason we can't use the normal edx-ccx-keys? https://pypi.python.org/pypi/edx-ccx-keys/0.2.1

dsjen replied:
I like this idea and tried it out, but unfortunately ran into a dependency error with the `six` python package.  I believe it should be a requirement of opaque keys because it's being used, https://github.com/edx/opaque-keys/blob/master/opaque_keys/__init__.py#L13.  However, it's not in the requirements,  https://github.com/edx/opaque-keys/blob/master/test-requirements.txt.

I'm asking around about it right now.  What do you think about a follow up upgrade once we get that sorted out?

dsjen replied:
Ok, problems resolved!  I've updated version numbers.

thallada replied:
Glad we got it sorted!

thallada commented on requirements/base.txt:
```diff
@@ -13,4 +13,6 @@ ordered-set==2.0.1                  # MIT
 Markdown==2.6.6    					# BSD
 
 -e git+https://github.com/edx/opaque-keys.git@d45d0bd8d64c69531be69178b9505b5d38806ce0#egg=opaque-keys
```
While we are at it, this package is also up on pypi: https://pypi.python.org/pypi/edx-opaque-keys/0.3.4

dsjen replied:
Version updated (to the brand new 0.4.0).

dsjen commented:
@thallada -- I added in more course_id tests for test_learners and added a health check timeout after creating the elasticsearch index to rectify the flakiness.  Ready for re-review!
